tools/wasi/wasi.cpp

```diff
@@ -0,0 +1,146 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2019-2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "execute.hpp"
+#include "parser.hpp"
+#include <uvwasi.h>
+#include <cassert>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+
+namespace fs = std::filesystem;
+
+namespace
+{
+// Global state.
+// NOTE: we do not care about uvwasi_destroy(), because it only clears file descriptors (currently)
+// and we are a single-run tool. This may change in the future and should reevaluate.
+uvwasi_t state;
+
+fizzy::execution_result wasi_proc_exit(fizzy::Instance&, std::vector<uint64_t> args, int)
```

Comment:
can do this and remove `return` statement
```suggestion
[[noreturn]] fizzy::execution_result wasi_proc_exit(fizzy::Instance&, std::vector<uint64_t> args, int)
```

Reply:
Ah no, doesn't work because of `-Winvalid-noreturn`

Reply:
You can try opening an issue on `uvwasi`.